### DISCUSSION TITLE
Next 13 app/ directory support

### DISCUSSION THREAD

**Discussion by HenrySkup** [tinacms/tinacms#3969] [Help]
Hi, I was wondering if there are any suggestions for use with the new Next app/ directory and server component based framework

**Comment by khnn** [comment]
Hey, so far I got it to work with the app router changing the default (tina init) code like this:

const getData = async (filename: string) => {
  let data = {}
  let query = {}
  let variables = { relativePath: `${filename}.mdx` }
  try {
    const res = await client.queries.post(variables)
    query = res.query
    data = res.data
    variables = res.variables
  } catch {
    // swallow errors related to document creation
  }

In [filename]/page.tsx you can then fetch and render `<TinaMarkdown />`

Haven't run into other big issues so far, but I might since it seems not fully supported.

A thing to consider is that useTina just works on the client side, so I might use a server rendered component for users and a client rendered one for admins to use visual editing.
